geocoding_platform_interface/lib/src/models/location.dart

```diff
@@ -55,14 +55,18 @@ class Location {
     }
 
     final Map<dynamic, dynamic> locationMap = message;
-    final timestamp = locationMap['timestamp'] != null
-        ? DateTime.fromMillisecondsSinceEpoch(locationMap['timestamp'].toInt(),
-            isUtc: true)
-        : null;
+    final timestamp = DateTime.fromMillisecondsSinceEpoch(
+        locationMap['timestamp'].toInt(),
+        isUtc: true);
+
+    if (locationMap['latitude'] == null && locationMap['longitude'] == null) {
```

Comment:
This should be an or: `||`. Since if one of the values (latitude or longitude) equals `null` we should throw an `ArgumentError` (we should also catch this in a test to prevent this mistake in the future).
```suggestion
    if (locationMap['latitude'] == null || locationMap['longitude'] == null) {
```